web/src/dropdown_widget.js

```diff
@@ -0,0 +1,134 @@
+                    // Let the browser handle it.
+                    return;
+                }
+
+                function first_item() {
+                    const first_item = list_items[0];
+                    return $popper.find(`.list-item[data-unique-id="${first_item.unique_id}"]`);
+                }
+
+                function last_item() {
+                    const last_item = list_items.at(-1);
+                    return $popper.find(`.list-item[data-unique-id="${last_item.unique_id}"]`);
+                }
+
+                switch (e.key) {
+                    case "Enter":
+                        if (list_items.length === 0) {
+                            $(e.target).trigger("click");
+                        } else {
+                            // Select first item.
+                            first_item().trigger("click");
+                        }
+                        e.target.click();
+                        e.stopPropagation();
+                        e.preventDefault();
+                        break;
+
+                    case "Escape":
+                        instance.hide();
+                        e.stopPropagation();
+                        e.preventDefault();
+                        break;
+
+                    case "Tab":
+                    case "ArrowDown":
+                        switch (e.target) {
+                            case last_item().get(0):
+                                trigger_element_focus($search_input);
+                                break;
+                            case $search_input.get(0):
+                                trigger_element_focus(first_item());
+                                break;
+                            default:
+                                trigger_element_focus($(e.target).next());
+                        }
+                        break;
+
+                    case "ArrowUp":
+                        switch (e.target) {
+                            case first_item().get(0):
+                                trigger_element_focus($search_input);
+                                break;
+                            case $search_input.get(0):
+                                // Can't focus on the last element since it is
+                                // most likely not rendered.
+                                break;
+                            default:
+                                trigger_element_focus($(e.target).prev());
+                        }
+                        break;
+                }
+            });
+
+            // Click on item.
+            $popper.one("click", ".list-item", (event) => {
+                item_click_callback(event, instance);
+            });
+
+            // Set focus on search input when dropdown opens.
+            setTimeout(() => {
+                $(".dropdown-list-search-input").trigger("focus");
+            });
```

Comment:
Hmm, I wonder if this could be done better as an on-show callback.

Reply:
I felt we want this behaviour for all the dropdowns, so it belongs here?